include/gridtools/stencil_composition/backend_naive/fused_mss_loop_naive.hpp

```diff
@@ -9,34 +9,94 @@
  */
 #pragma once
 
+#include <cstdlib>
+
+#include "../../common/generic_metafunctions/for_each.hpp"
+#include "../../common/tuple_util.hpp"
 #include "../../meta.hpp"
-#include "../mss_functor.hpp"
+#include "../grid.hpp"
+#include "../local_domain.hpp"
+#include "../pos3.hpp"
+#include "iterate_domain_naive.hpp"
 
-/**@file
- * @brief mss loop implementations for the x86 backend
- */
 namespace gridtools {
-    /**
-     * @brief struct holding backend-specific runtime information about stencil execution.
-     */
-    struct execution_info_naive {};
+    namespace naive_impl_ {
+        template <class LocalDomain, class Grid>
+        struct stage_executor_f {
+            LocalDomain const &m_local_domain;
+            Grid const &m_grid;
+
+            template <template <class...> class L, class From, class To, class Stage>
+            void operator()(L<From, To, Stage>) const {
+                using extent_t = typename Stage::extent_t;
+                int_t iminus = extent_t::iminus::value;
+                int_t iplus = extent_t::iplus::value;
+                int_t jminus = extent_t::jminus::value;
+                int_t jplus = extent_t::jplus::value;
+                int_t k_from = m_grid.template value_at<From>();
+                int_t k_to = m_grid.template value_at<To>();
+                int_t i_count = m_grid.i_high_bound() - m_grid.i_low_bound() + 1 + iplus - iminus;
+                int_t j_count = m_grid.j_high_bound() - m_grid.j_low_bound() + 1 + jplus - jminus;
+                int_t k_count = 1 + std::abs(k_to - k_from);
+                int_t k_step = k_to >= k_from ? 1 : -1;
+
+                iterate_domain_naive<LocalDomain> it_domain(m_local_domain, m_grid);
+
+                it_domain.increment_i(iminus);
```

Comment:
I think some comments would be useful here. The naive implementation can then serve as an example for new backend developers.

Reply:
done